
        
        
@functools.lru_cache()
def get_hstore_oids(connection_alias):
    '''Return hstore and hstore array OIDs.'''
    with connections[connection_alias].cursor() as cursor:
        cursor.execute(
            'SELECT t.oid, typarray '
            'FROM pg_type t '
            'JOIN pg_namespace ns ON typnamespace = ns.oid '
            'WHERE typname = 'hstore''
        )
        oids = []
        array_oids = []
        for row in cursor:
            oids.append(row[0])
            array_oids.append(row[1])
        return tuple(oids), tuple(array_oids)
    
    
class RangeMinValueValidator(MinValueValidator):
    def compare(self, a, b):
        return a.lower is None or a.lower < b
    message = _('Ensure that this range is completely greater than or equal to %(limit_value)s.')

    
    
class RedirectFallbackMiddleware(MiddlewareMixin):
    # Defined as class-level attributes to be subclassing-friendly.
    response_gone_class = HttpResponseGone
    response_redirect_class = HttpResponsePermanentRedirect
    
        def exists(self, session_key=None):
        '''
        This method makes sense when you're talking to a shared resource, but
        it doesn't matter when you're storing the information in the client's
        cookie.
        '''
        return False
    
    for face_location in face_locations:
    
    for face_location in face_locations:
    
        if os.path.isdir(image_to_check):
        if cpus == 1:
            [test_image(image_file, model) for image_file in image_files_in_folder(image_to_check)]
        else:
            process_images_in_process_pool(image_files_in_folder(image_to_check), cpus, model)
    else:
        test_image(image_to_check, model)
    
    # Load some sample pictures and learn how to recognize them.
lmm_image = face_recognition.load_image_file('lin-manuel-miranda.png')
lmm_face_encoding = face_recognition.face_encodings(lmm_image)[0]
    
            if self.flash_briefings.get(briefing_id) is None:
            err = 'No configured Alexa flash briefing was found for: %s'
            _LOGGER.error(err, briefing_id)
            return b'', 404
    
    from homeassistant.core import callback
from homeassistant.const import CONF_PLATFORM
import homeassistant.helpers.config_validation as cv
import homeassistant.util.dt as dt_util
from homeassistant.helpers.event import track_point_in_utc_time
    
    
class UserNotConfirmed(CloudError):
    '''Raised when a user has not confirmed email yet.'''
    
                    _LOGGER.debug('Device %s is connected', mac)
                self.last_results[mac] = name
        except (KeyError, IndexError):
            _LOGGER.exception('Router returned unexpected response')
            return False
        return True
    
        async def async_update_callback(self):
        '''Update the device.'''
        await self.async_update()
        node = self.gateway.sensors[self.node_id]
        child = node.children[self.child_id]
        position = child.values[self.value_type]
        latitude, longitude, _ = position.split(',')